### DISCUSSION TITLE
Missing required parameter for [Route: filament.app.pages.edit-profile] [URI: app/team/{tenant}/edit-profile] [Missing parameter: tenant].

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#12019] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.59

### How can we help you?

<img width="1363" alt="Screenshot 2024-03-24 at 13 52 11" src="https://github.com/filamentphp/filament/assets/121255432/8204ff6b-b4a1-43c3-a9ec-612c48a76519">

```
->userMenuItems([
    MenuItem::make()
        ->icon('heroicon-o-user-circle')
        ->label(fn (): string => EditProfile::getLabel())
        ->url(fn () => EditProfile::getUrl(parameters: [
            'tenant' => Filament::getTenant(),
        ])),
])
```

**Comment by leandrocfe** [comment]
```php
EditProfile::getUrl(['tenant' => Filament::getTenant()->id ?? 0])
```
?

**Reply by devhoussam1998** [reply]
as you can see in my code below I'm using the slug.

`->url(fn () => EditProfile::getUrl(['tenant' => Filament::getTenant()->slug ?? 0]))`

https://github.com/filamentphp/filament/assets/121255432/1025fd67-9d99-40b3-92ef-85468c58609f

**Comment by micraux** [comment]
I get the same error in a controller: seems `Filament::getTenant()` is returning null.

@devhoussam1998, did you manage to solve it?

If not, try `dd(Filament::getTenant())` to see what is being returned.

**Reply by devhoussam1998** [reply]
I still not find the correct solution for this

**Reply by micraux** [reply]
I did manage to solve the problem by creating a `RememberTenant` middleware in order to store the current tenant in session.

I chose this way because I also want to be able to store the last tenant in the DB in order to automatically connect the user to the tenant he was working on when he logged out: in my projet, I have a many-to-many relationship for Users and Tenants.

**Reply by devhoussam1998** [reply]
could you share your solution

**Comment by micraux** [comment]
Here you are...

1. Create a middleware
```php
<?php

namespace App\Http\Middleware;

use Closure;
use Filament\Facades\Filament;
use Illuminate\Http\Request;
use Symfony\Component\HttpFoundation\Response;

class RememberTenant
{
    /**
     * Handle an incoming request.
     *
     * @param  \Closure(\Illuminate\Http\Request): (\Symfony\Component\HttpFoundation\Response)  $next
     */
    public function handle(Request $request, Closure $next): Response
    {
        $request->session()->put('tenant_id', Filament::getTenant()->id);

        return $next($request);
    }
}
```

2. Declare the middleware in your panel
```php
->tenantMiddleware(
    [
        ...
        RememberTenant::class,
    ],
    isPersistent: true
)
```

**Comment by devhoussam1998** [comment, marked as answer]
here is what I did it works

  ```
  public function getCancelFormAction(): Action
    {
        return Action::make('cancel')
            ->label(__('Cancel'))
            ->size(ActionSize::Large)
            ->color('gray')
            ->icon('heroicon-o-x-circle')
            ->visible(fn (): bool => auth()->user()?->directories()->exists())
            ->alpineClickHandler('document.referrer ? window.history.back() : (window.location.href = ' . Js::from(Dashboard::getUrl(['tenant' => Filament::getTenant() ?? auth()->user()?->directories()->first()?->slug], panel: 'app')) . ')');
    }
```
